core/drivers/gic.c

```diff
@@ -948,11 +950,34 @@ static void __maybe_unused gic_native_itr_handler(void)
 	iar = gic_read_iar(gd);
 	id = iar & GICC_IAR_IT_ID_MASK;
 
-	if (id <= gd->max_it)
+	if (id <= gd->max_it) {
 		interrupt_call_handlers(&gd->chip, id);
-	else
-		DMSG("ignoring interrupt %" PRIu32, id);
+		goto write_eoir;
+	} else if (IS_ENABLED(CFG_ARM_GICV3) && id >= 1020 && id <= 1023) {
```

Comment:
It's getting hard to follow all possible flows. The `else` isn't needed. However, I think it would be even better to handle the special interrupts before "if (id <= gd->max_it)" since we're going to return early from the function for those.

Reply:
Yes indeed, I'll come up with something clearer while applying your suggestion